Stetskyi_Homework_13/ADOClassLibrary/ADOClassLibrary/PerformerFolder/Performer.cs

```diff
@@ -0,0 +1,559 @@
+            {
+                SqlCommand cmd = new SqlCommand(@"EXEC InitializeDB", connection);
+                try
+                {
+                    connection.Open();
+                    cmd.Transaction = connection.BeginTransaction(IsolationLevel.Serializable);
+                    cmd.ExecuteNonQuery();
+                    cmd.Transaction.Commit();
+                }
+                catch (Exception e)
+                {
+                    cmd.Transaction.Rollback();
+                    throw new Exception($"Failed to execute Stored Procedure, \nReason: {e.Message}");
+                }
+            }
+        }
+        //Disconnected Architecture
+        public List<Order> GetAllOrders()
+        {
+            List<Order> orders = new List<Order>();
+            using (SqlConnection connection = connectionProvider.GetConnection())
+            {
+                SqlCommand cmd = new SqlCommand(@"select
+                                                    [Id], 
+                                                    [Status], 
+                                                    [CreatedDate],
+                                                    [UpdatedDate],
+                                                    [ProductId] from Orders", connection);
+
+                SqlDataAdapter sda = new SqlDataAdapter(cmd);
+                DataSet dsData = new DataSet();
+                sda.Fill(dsData);
+
+                for (int i = 0; i < dsData.Tables[0].Rows.Count; i++)
+                {
+                    Order o = new Order();
+                    o.Id = Convert.ToInt32(dsData.Tables[0].Rows[i]["Id"]);
+                    o.Status = Convert.ToString(dsData.Tables[0].Rows[i]["Status"]);
+                    o.CreatedDate = Convert.ToDateTime(dsData.Tables[0].Rows[i]["CreatedDate"]);
+                    o.UpdatedDate = Convert.ToDateTime(dsData.Tables[0].Rows[i]["UpdatedDate"]);
+                    o.ProductId = Convert.ToInt32(dsData.Tables[0].Rows[i]["ProductId"]);
+                    orders.Add(o);
+                }
+                return orders;
+            }
+        }
+        //Connected Architecture
+        public List<Product> GetAllProducts()
+        {
+            List<Product> products = new List<Product>();
+
+            using (SqlConnection connection = connectionProvider.GetConnection())
+            {
+                SqlCommand cmd = new SqlCommand(@"select [Id],
+                                                         [Description],
+                                                         [Weight],
+                                                         [Height],
+                                                         [Width],
+                                                         [Length] from Products", connection);
+                SqlDataReader reader;
+                try
+                {
+                    connection.Open();
```

Comment:
@IhorStetskyi, we could extract the same flow in a separate method and reuse it in needed places.